### PR TITLE
Update profiler_caliper.f90

### PR DESCRIPTION
**Pull request by mathrack**
Remove unused variables to allow `dev` builds

### CODE REVIEW

File: src/profiler_caliper.f90 [hunk 1]
```diff
@@ -127,9 +124,9 @@ subroutine manager_error(mmm)
       logical :: ret
       character(len=:), allocatable :: errmsg
 
-      ret = mgr%error()
+      ret = mmm%error()
```
Comment (pbartholomew08) [inline src/profiler_caliper.f90:127]: Why not call `mmm` -> `mgr`? It would be easier to read I think.

**Review by pbartholomew08 — CHANGES_REQUESTED** [review]

File: src/profiler_caliper.f90 (hunk 1 above)
Comment (pbartholomew08) [reply src/profiler_caliper.f90:127]: Or if concerned `mgr` might clash with the module-level variable, another name than `mmm` would be prefereable

Comment (mathrack) [reply src/profiler_caliper.f90:127]: Sounds good, I will use `manager`

Comment (mathrack) [reply src/profiler_caliper.f90]: Fixed

**Review by pbartholomew08 — APPROVED** [review]
LGTM!

**Pull request merged**